### PR TITLE
Migration tests

### PR DESCRIPTION
**Pull request by stu-gott**
The test suite currently fails but I believe it should work once all feature branches are merged.

### CODE REVIEW

**Comment by stu-gott** [comment]
I'm not entirely sure I like the way getCoreRESTClient was implemented. @rmohr what do you think?

**Comment by rmohr** [comment]
@stu-gott you can do `GetFromFlags(...).CoreV1().RESTClient()` instead. No extra configuration necessary for it.

**Comment by stu-gott** [comment]
I thought there was a better way. Thank you!

**Comment by stu-gott** [comment]
@rmohr @admiyo @fabiand This is ready for another look

File: tests/vm_migration_test.go [hunk 1]
```diff
@@ -140,6 +142,30 @@ var _ = Describe("VmMigration", func() {
...
+			vm, err := restClient.Post().Resource("vms").Namespace(k8sv1.NamespaceDefault).Body(sourceVM).Do().Get()
+			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMStart(vm)
+
+			// Create the Migration
+			err = restClient.Post().Resource("migrations").Namespace(k8sv1.NamespaceDefault).Body(migration).Do().Error()
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() bool {
+				obj, err := coreRestClient.Get().Resource("pods").Namespace(k8sv1.NamespaceDefault).Do().Get()
```
Comment (rmohr) [inline tests/vm_migration_test.go]: You can do a selector and directly select the pod. The query should be 

```golang
v1.AppLabel + "=migration," + v1.MigrationUID + "=" + migration.GetObjectMeta().GetUID()
```

**Comment by rmohr** [comment]
@admiyo please first investigate why CI fails before merging ...

**Pull request merged**